src/scripts/main.js

```diff
@@ -1,3 +1,43 @@
 'use strict';
 
-// write code here
+'use strict';
+
+const employeeList = document.querySelector('ul');
+
+const parseSalary = (salaryStr) => {
+  let cleanStr = salaryStr.split('$')[1];
+
+  cleanStr = cleanStr.split(',').join('');
+
+  return parseFloat(cleanStr);
+};
+
+const sortEmployees = (list) => {
+  const employeeArray = Array.from(list.children);
+
+  const sortedEmployees = employeeArray.sort((a, b) => {
+    const salaryA = parseSalary(a.getAttribute('data-salary'));
+    const salaryB = parseSalary(b.getAttribute('data-salary'));
+
+    return salaryB - salaryA;
+  });
+
+  list.innerHTML = '';
+  sortedEmployees.forEach((employee) => list.appendChild(employee));
+};
+
+const getEmployeeData = (list) => {
+  const employeeArray = Array.from(list.children);
+
+  return employeeArray.map((employee) => {
+    return {
+      name: employee.textContent.trim(),
+      position: employee.getAttribute('data-position'),
+      salary: parseSalary(employee.getAttribute('data-salary')),
+      age: parseInt(employee.getAttribute('data-age')),
```

Comment:
Consider adding error handling for parseInt to ensure that the age attribute is a valid number. This will prevent potential NaN values if the attribute is missing or malformed.